### PR TITLE
Use RefLike for Remote.name

### LINKED ISSUE #414: Use RefLike for Remote name
Currently the [`name` field](https://github.com/radicle-dev/radicle-link/blob/next/librad/src/git/types/remote.rs#L26) for a `Remote` is of type `String`. For stronger guarantees of safety, we should migrate this to use `RefLike`.

One place this will affect will be `include.rs` where we use `Remote`s to build include files for remote fetching. See [this conversation](https://github.com/radicle-dev/radicle-link/pull/412#discussion_r520633739) for some further detail.

### PR DESCRIPTION
**Pull request by FintanH**
Fixes #414 

I'm not gonna backwards patch this because it's not critical.

### CODE REVIEW

File: librad/src/git/include.rs [hunk 1]
```diff
@@ -149,22 +147,25 @@ impl<Path> Include<Path> {
...
     ///
     /// The tracked users are expected to be retrieved by talking to the
     /// [`crate::git::storage::Storage`].
     #[tracing::instrument(level = "debug")]
-    pub fn from_tracked_users<I>(path: Path, local_url: LocalUrl, tracked: I) -> Result<Self, Error>
+    pub fn from_tracked_users<R, I>(
+        path: Path,
+        local_url: LocalUrl,
+        tracked: I,
+    ) -> Result<Self, Error>
```
Comment (kim) [inline librad/src/git/include.rs:159]:
```suggestion
    ) -> Self
```

File: librad/src/git/include.rs [hunk 2]
```diff
@@ -174,12 +175,13 @@ impl<Path> Include<Path> {
         })
     }
 
-    fn build_remote(url: LocalUrl, peer: PeerId, handle: &str) -> Result<Remote<LocalUrl>, Error> {
-        let name = format!("{}@{}", handle, peer);
+    fn build_remote(url: LocalUrl, peer: PeerId, handle: ext::RefLike) -> Remote<LocalUrl> {
```
Comment (kim) [inline librad/src/git/include.rs:178]:
```suggestion
    fn build_remote(url: LocalUrl, peer: PeerId, handle: impl Into<ext::RefLike>) -> Remote<LocalUrl> {
```

File: librad/src/git/include.rs [hunk 3]
```diff
@@ -94,10 +93,9 @@ impl<Path> Include<Path> {
         }
     }
 
-    pub fn add_remote(&mut self, url: LocalUrl, peer: PeerId, handle: &str) -> Result<(), Error> {
-        let remote = Self::build_remote(url, peer, handle)?;
+    pub fn add_remote(&mut self, url: LocalUrl, peer: PeerId, handle: ext::RefLike) {
```
Comment (kim) [inline librad/src/git/include.rs:96]:
```suggestion
    pub fn add_remote(&mut self, url: LocalUrl, peer: PeerId, handle: impl Into<ext::RefLike>) {
```

**Review by kim — APPROVED** [review]

File: librad/src/git/include.rs (hunk 1 above)
Comment (FintanH) [reply librad/src/git/include.rs:159]: Thanks, lost this while translating over from `maint` :sweat_smile:

**Review by kim — APPROVED** [review]

**Pull request merged**